### DISCUSSION TITLE
[4.0] Simpler configuration. Defaults setup. [suggestion]

### DISCUSSION THREAD

**Discussion by mrownicki** [joomla/joomla-cms#36768] [2. Cold Features]
### Steps to reproduce the issue
Joomla 4 must be a more friendly configuration for start. (Fresh, fast and simpler)
No need login to ftp and change htaccess.txt to .htaccess This is not friendly approach.
Kerwords Google not using. 90% people using google maybe more.
If we're using some defaults options and they are obvious that they will always be on, there is no need to show them.

### Expected result

Joomla after install start in .htaccess
Search Engine Friendly URLs - Yes (defaults) but remove
Use URL Rewriting - Yes (defaults) but remove
Send mail - YES (defaults) but remove.
FTP - remove completly.
Gzip - YES (defaults) or Brotli YES (defaults) 
Force https - Only administrator (REMOVE) change to Yes on NOT
Keywords - Remove 
Autor meta tag - move to article options
Joomla version - OFF (defaults) but remove
Cookie setting - I don't know if anyone uses it. (if no need remove)

### Additional comments
Joomla must be better and simpler for SEO and configuration. Newbie user expects working simple solution. 
I go further and don't show this options.
This in my option to bring more good for newbie user who install and immediately write article. They don't want to play setting the mass of options because they don't have time for it.
What do you think about this?

If you don't hear about brotli https://medium.com/oyotech/how-brotli-compression-gave-us-37-latency-improvement-14d41e50fee4

![screen shot 2019-10-31 at 11 52 19](https://issues.joomla.org/uploads/1/bc6a47d988eb926367b4caf32bff8fbf.png)![screen shot 2019-10-31 at 11 52 21](https://issues.joomla.org/uploads/1/2fe88aca7adefdda08743b5795d22405.png)![screen shot 2019-10-31 at 11 52 23](https://issues.joomla.org/uploads/1/76f7841601505190bd18059b644d24ea.png)![screen shot 2019-10-31 at 11 52 24](https://issues.joomla.org/uploads/1/5a3a2c8f693b5aaf04add9d1311041e8.png)![screen shot 2019-10-31 at 11 52 26](https://issues.joomla.org/uploads/1/c0b0eeb6d343eaca9d2ddde86400c5a8.png)![screen shot 2019-10-31 at 11 54 06](https://issues.joomla.org/uploads/1/bc3b95522e3590028ab4a14d21389c69.png)![screen shot 2019-10-31 at 11 54 57](https://issues.joomla.org/uploads/1/3dbdeb4b57e285ab70531297f5327152.png)

**Comment by mbabker** [comment]
Joomla supports more web servers than Apache.  Blindly renaming the htaccess.txt file won’t work for IIS or nginx.

Joomla’s default .htaccess does not work on all hosting providers (usually because of an issue with that host), blindly renaming the file would cause more issues than it solves.

Removing the option to disable mail sending makes no sense.  There are valid reasons for defaulting it on and allowing a site admin to turn it off.

The URL rewriting options must be used in conjunction with web server configuration, it is not “turn this on in Joomla” or “turn this on at the web server level”.  Also keep in mind it allows use of human friendly URLs even if the web server isn’t configured in the right way (you just have URLs like `/index.php/foo/bar`).

Cookie settings is an advanced thing.  But it should not be flat out removed.

**Comment by mrownicki** [comment]
Joomla not need support all hosting provider. Come on.
If we require **J4 php 7.3** then why can't we require mod_rewrite in Apache ??

No remove **mass maill** only send mail ;) If you have infection provider hosting cut off phpsendmail in 99% case.

Nobody need /index.php 

_Joomla’s default .htaccess does not work on all hosting providers (usually because of an issue with that host), blindly renaming the file would cause more issues than it solves._

Maybe we need fix that. I don't write blind ;) In most case mod_rewrite must be on. Simple. Sorry, but I didn't meet a provider who wouldn't offer a rewrite module on apache.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26876">issues.joomla.org/tracker/joomla-cms/26876</a>.</sub>

**Comment by mbabker** [comment]
> Joomla not need support all hosting provider. Come on.

Joomla is designed to be able to run on multiple web server platforms; this includes Apache, NGINX, IIS, and LiteSpeed.  Not all of these platforms utilize .htaccess for configuration directives and it is pretty narrow sighted if you're suggesting that every Joomla user must run their site on an Apache server.

> If we require **J4 php 7.3** then why can't we require mod_rewrite in Apache ??

URL rewriting is a 100% optional feature and not necessary for Joomla to function properly.  To my knowledge this module being enabled is only a hard requirement for use of Drupal, other CMS's I use or have evaluated do not impose this requirement.

>> _Joomla’s default .htaccess does not work on all hosting providers (usually because of an issue with that host), blindly renaming the file would cause more issues than it solves._
> 
> Maybe we need fix that.

It has more to do with whether mod_rewrite is available or not.  The default file in 4.0 wraps everything that can be inside of conditionals to determine if a feature is available, but there is still the `Options` directive which may be disabled for whatever reason or a user needing to adjust the `RewriteBase` directive for proper URL rewriting in some scenarios.  The point here is that Joomla as a web application is not in a spot to blindly turn on a web server configuration file and if it breaks be unable to help users fix that issue (because if the `.htaccess` is borked then you're not getting into the UI at all in most cases).

Maybe you and I run all of our sites in optimal server configurations, and maybe you only use Apache for your web server of choice, but there are a lot more server configurations out there and a "you must have an optimal server configuration to use Joomla" approach is kind of disrespectful to those who aren't technically savvy.

**Comment by mrownicki** [comment]
https://prnt.sc/pqsc2u
Apache, Ngnix 70% world usage. (lightspeed, ISS, Cloud other 25%) = 70+25=95% this is still very good. 
I don't agree this is approach is kind of disrespectful. This is smart. 
Why joomla does not support plugins from version 1.5 This is so disrespectful for develeopers ;) (There is no support to get rid of technological debt.)
Joomla does not have to run on every server. It is supposed to run on the most popular ones.
An opportunity arises to give up unused technologies and change something with J4, why not do it.
Not all Joomla users are programmers ... and many people get the impression that joomla is too difficult and written by programmers for programmers.

Joomla using in the world drop in 3.0% to 2.7% (in 2019)

Please, think to gain in the market and not lose. :(
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26876">issues.joomla.org/tracker/joomla-cms/26876</a>.</sub>

**Comment by Hackwar** [comment]
I'm sorry, but @mbabker is right here. We can't simply assume everybody is using Apache and we can't force everybody to use mod_rewrite SEF URLs, etc. The simple fact that there are thousands of websites out there without mod_rewrite URLs and we would invalidate all their URLs by forcing them to use mod_rewrite, makes this a non-starter. Yes, we can improve Joomla quite a bit, but your proposals aren't things we can do.
